server/src/weaverbird/backends/mongo_translator/steps/formula.py

```diff
@@ -1,128 +1,6 @@
-    else:
-        raise InvalidFormula(f'Operator {unop.op.__class__} is not supported')
-
-
-def mongo_formula_for_ast_node(node: ast.AST, columns_aliases: Dict) -> Any:
-    if isinstance(node, ast.BinOp):
-        return mongo_formula_for_binop(node, columns_aliases)
-    elif isinstance(node, ast.UnaryOp):
-        return mongo_formula_for_unaryop(node, columns_aliases)
-    elif isinstance(node, ast.Constant):
-        return mongo_formula_for_constant(node)
-    elif isinstance(node, ast.Name):
-        return mongo_formula_for_name(node, columns_aliases)
-    else:
-        raise InvalidFormula(f'Formula node {node} is not supported')
-
-
-def mongo_formula_for_expr(expr: ast.Expr, columns_aliases: Dict) -> Dict[str, Any]:
-    if isinstance(expr.value, ast.AST):
-        return mongo_formula_for_ast_node(expr.value, columns_aliases)
-    else:
-        raise InvalidFormula
+from weaverbird.backends.utils.formula_builder import MongoFormulaBuilder
+from weaverbird.pipeline.steps.formula import FormulaStep
 
 
 def translate_formula(step: FormulaStep) -> list:
-    return [{'$addFields': {step.new_column: build_mongo_formula_tree(step.formula)}}]
-
-
-def build_mongo_formula_tree(
-    formula: Union[str, int, float, bool]
-) -> Union[Dict[str, Any], int, float, bool]:
-    if isinstance(formula, str):
-        columns_aliases, sanitized_formula = sanitize_formula(formula)
-        module = ast.parse(sanitized_formula)
-        expr = module.body[0]
-        if not isinstance(expr, ast.Expr):
-            raise InvalidFormula
-        mongo_expr = mongo_formula_for_expr(expr, columns_aliases)
-        return mongo_expr
-    elif type(formula) in (int, float, bool):
-        return formula
-    else:
-        raise InvalidFormula
-
-
-def sanitize_formula(formula: str) -> Tuple[Dict[str, str], str]:
-    """
-    This function handles column names with special characters & spaces.
-    Such columns are surrounded by brackets. The functions replaces this pattern by a temporary name
-    and store the original name in a dict, with the temporary name as key and the actual name as key.
-    For example in this formula -> [I'm a Special Column!!] * 10 it will return
-    tuple(__vqb_col_0__ * 10, {"__vqb_col_0__": "[I'm a Special Column!!]"})
-    The translate function will replaces the temporary name by the old one once the formula will
-    be parsed by ast.parse and translated to mongo.
-    """
-    columns_aliases = {}
-    sanitized_formula = formula
-    matches = re.findall(r'\[(.*?)]', formula)
-    for i, match in enumerate(matches):
-        columns_aliases[f'__vqb_col_{i}__'] = match
-        sanitized_formula = sanitized_formula.replace(f'[{match}]', f'__vqb_col_{i}__')
-    return columns_aliases, sanitized_formula
-
-
-class InvalidFormula(Exception):
-    """
-    Raised when a formula is not supported
-    """
+    return MongoFormulaBuilder.translate_formula(step)
```

Comment:
Maybe we should rename this to something more generic.

Or create a new specifique class for SQL

Reply:
mhhh... not sure to understand well, can you provide an example of your renaming proposition please ?

Reply:
```suggestion
    return FormulaBuilder.translate_formula(step)
```